HW0/P4/P4.py

```diff
@@ -30,47 +30,94 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    s_true = np.loadtxt('P4_trajectory.txt', delimiter=',')
+    x_coords = [sk[0] for sk in s_true]
+    y_coords = [sk[1] for sk in s_true]
+    z_coords = [sk[2] for sk in s_true]
+    ax.plot(x_coords, y_coords, z_coords,
+            '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
+    ms = np.loadtxt('P4_measurements.txt', delimiter=',')
+    mat = np.matrix([[1.0/rx, 0, 0], [0, 1.0/ry, 0], [0, 0, 1.0/rz]])
+    xs = ms*mat
+    xs = np.array(xs)
+    x_coords = [x[0] for x in xs]
+    y_coords = [x[1] for x in xs]
+    z_coords = [x[2] for x in xs]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.array([[1,0,0,dt,0,0], [0,1,0,0,dt,0], [0,0,1,0,0,dt], [0,0,0,1-c*dt,0,0], [0,0,0,0,1-c*dt,0], [0,0,0,0,0,1-c*dt]])
```

Comment:
Easy to infer matrix based on rows but perhaps can split definition into 6 lines, one row/line using "\" I believe, to make it even easier to read.